Dockerfile

```diff
@@ -1,73 +1,206 @@
+    # Start ganache-cli in background with specific host and port
+    nohup ganache-cli \
+        --host 0.0.0.0 \
+        --port 8545 \
+        --networkId 1234 \
+        --accounts 10 \
+        --defaultBalanceEther 1000 \
+        --mnemonic "test test test test test test test test test test test junk" \
+        --db /tmp/ganache_db \
+        > ganache.log 2>&1 & \
+    # Store PID and wait
+    GANACHE_PID=$! && \
+    echo "Started Ganache with PID: $GANACHE_PID" && \
+    sleep 15 && \
+    # Debug: show ganache logs
+    echo "Ganache logs:" && \
+    cat ganache.log && \
+    # Debug: check if process is running
+    ps aux | grep ganache && \
+    # Verify ganache is running
+    curl -v http://localhost:8545 && \
+    # Continue with build
+    rm -rf build && \
+    ./build.sh && \
+    mkdir -p build/Release && \
+    mkdir -p starkgate-artifacts && \
+    cp -r build/Release/src/* starkgate-artifacts/ && \
+    # Kill ganache after build
+    kill $GANACHE_PID || true
+
+# Remove existing Node.js and related packages
+RUN apt-get purge -y nodejs nodejs-doc node-gyp libnode-dev && \
+    apt-get autoremove -y && \
+    rm -rf /etc/apt/sources.list.d/nodesource.list && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install Node.js 18.x
+RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
+    apt-get update && \
+    apt-get install -y nodejs && \
+    # Install a specific compatible version of npm
+    npm install -g npm@9.8.1
+
+# Install and verify ganache with explicit path and shell
+RUN npm install -g --unsafe-perm ganache@7.9.0 && \
+    # Verify ganache installation and keep trying if it fails
+    (for i in {1..5}; do \
+        if which ganache && ganache --version; then \
+            echo "Ganache installation verified" && \
+            break; \
+        else \
+            echo "Attempt $i: Ganache not found, trying again..." && \
+            echo "Searching for ganache in PATH..." && \
+            find / -name ganache 2>/dev/null && \
+            # Clear npm cache and reinstall
+            npm cache clean --force && \
+            npm install -g --unsafe-perm ganache@7.9.0 && \
+            # Add environment variables
+            export PATH="/usr/local/lib/node_modules/.bin:$PATH" && \
+            # Add a small delay to let npm finish
+            sleep 5; \
+        fi; \
+        if [ $i -eq 5 ]; then \
+            echo "Failed to install ganache after 5 attempts" && \
+            exit 1; \
+        fi \
+    done)
+
+# Generate other artifacts
+RUN . "$HOME/.asdf/asdf.sh" && \
+    ls -la /app/.cairo && \
+    ls -la /app/.cairo/cairo/bin && \
+    export PATH="/app/.cairo/cairo/bin:$PATH" && \
+    echo $PATH && \
+    which starknet-compile && \
+    make starkgate-contracts-latest && \
+    make braavos-account-cairo && \
+    make argent-contracts-starknet
+
+# Build the Rust project with specific binary name
+RUN cargo build --release --workspace --bin karnot-bridge-deploy
```

Comment:
let's change the binary name also to madara-bootstrapper, looks like we forgot to change this long before

Reply:
done